server/concept/TypeService.java

```diff
@@ -101,339 +110,357 @@ public class TypeService {
     private final TransactionService transactionSvc;
     private final ConceptManager conceptMgr;
 
+    private final EnumMap<ConceptProto.RoleType.Req.ReqCase, BiConsumer<ConceptProto.RoleType.Req, UUID>> role_type_handlers;
+    private final EnumMap<ConceptProto.ThingType.Req.ReqCase, BiConsumer<ConceptProto.ThingType.Req, UUID>> thing_type_handlers;
+
     public TypeService(TransactionService transactionSvc, ConceptManager conceptMgr) {
         this.transactionSvc = transactionSvc;
         this.conceptMgr = conceptMgr;
+
```

Comment:
Ideally, since they are so distinct, we'd have a `RoleTypeService` and a `ThingTypeService`. Keeping as a single `TypeService` for the time being.

Reply:
Another option would be to merge role type requests into relation type requests.